### ISSUE TITLE
Feature Request: Open file through Asset Processor

### ISSUE THREAD

**Issue by ParasiteSt** [o3de/o3de#11512]
**Is your feature request related to a problem? Please describe.**
Its not a huge problem, but it allows for easy opening of files that fail if they are able to be opened in o3de, and could save developers a lot of time.

**Describe the solution you'd like**
If some file fails to process in the AP, if it is open-able in o3de, you should be able to double click on the source in the AP and it should open the file in o3de in the proper window. For example, if I have a failed script canvas graph, and see this in the AP, right now you have to see the name of the file/s, and go into the script canvas tool and open the graphs up, the solution would be to double click the Source and it bring you to that exact same point without having to jump through hoops to get there.

**Describe alternatives you've considered**
An alternative is possibly a right click menu button that would open the file in editor. This alternative could be a better route as the right click menu already has other open with buttons, so adding one more could be an easier and possible better alternative as it would stop any accidental openings of the file that you could get with a double click approach.

This is the right click menu right now in the AP.
![image](https://user-images.githubusercontent.com/100494538/186314700-e63b5b12-e825-4f8f-a06d-9e86ea5fc29c.png)

**Additional context**
After going through both solutions I feel the right click menu alternative may be a better approach, but having others discuss this will be needed to come up with the best solution, if this FR gets to that point.

**Comment by lmbr-pip** [comment]
This would need some understanding on a per asset type to see what the right behaviour should be, sometimes in editor you are working with product assets, sometimes you work with source assets. If source assets fail to process, its unclear what the right action would be for some asset types.

**Comment by ParasiteSt** [comment]
Yes I completely agree, I am mainly talking for product assets, and not as much source assets. This Feature Request is more of an alternative solution to help with this issue (https://github.com/o3de/o3de/issues/10211).
